src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratedDocumentState.cs

```diff
@@ -7,120 +7,136 @@
-            ParseOptions parseOptions,
-            LanguageServices languageServices)
-        {
-            var loadTextOptions = new LoadTextOptions(generatedSourceText.ChecksumAlgorithm);
-            var textAndVersion = TextAndVersion.Create(generatedSourceText, VersionStamp.Create());
-            var textSource = new ConstantTextAndVersionSource(textAndVersion);
-            var treeSource = CreateLazyFullyParsedTree(
-                textSource,
-                loadTextOptions,
-                documentIdentity.FilePath,
-                parseOptions,
-                languageServices);
-
-            return new SourceGeneratedDocumentState(
-                documentIdentity,
-                languageServices,
-                documentServiceProvider: SourceGeneratedTextDocumentServiceProvider.Instance,
-                new DocumentInfo.DocumentAttributes(
-                    documentIdentity.DocumentId,
-                    name: documentIdentity.HintName,
-                    folders: SpecializedCollections.EmptyReadOnlyList<string>(),
-                    parseOptions.Kind,
-                    filePath: documentIdentity.FilePath,
-                    isGenerated: true,
-                    designTimeOnly: false),
-                parseOptions,
-                textSource,
-                loadTextOptions,
-                treeSource);
-        }
+        var loadTextOptions = new LoadTextOptions(generatedSourceText.ChecksumAlgorithm);
+        var textAndVersion = TextAndVersion.Create(generatedSourceText, VersionStamp.Create());
+        var textSource = new ConstantTextAndVersionSource(textAndVersion);
+        var treeSource = CreateLazyFullyParsedTree(
+            textSource,
+            loadTextOptions,
+            documentIdentity.FilePath,
+            parseOptions,
+            languageServices);
+
+        return new SourceGeneratedDocumentState(
+            documentIdentity,
+            languageServices,
+            documentServiceProvider: SourceGeneratedTextDocumentServiceProvider.Instance,
+            new DocumentInfo.DocumentAttributes(
+                documentIdentity.DocumentId,
+                name: documentIdentity.HintName,
+                folders: SpecializedCollections.EmptyReadOnlyList<string>(),
+                parseOptions.Kind,
+                filePath: documentIdentity.FilePath,
+                isGenerated: true,
+                designTimeOnly: false),
+            parseOptions,
+            textSource,
+            generatedSourceText,
+            loadTextOptions,
+            treeSource);
+    }
+
+    private SourceGeneratedDocumentState(
+        SourceGeneratedDocumentIdentity documentIdentity,
+        LanguageServices languageServices,
+        IDocumentServiceProvider? documentServiceProvider,
+        DocumentInfo.DocumentAttributes attributes,
+        ParseOptions options,
+        ConstantTextAndVersionSource textSource,
+        SourceText text,
```

Comment:
i looked into this, and it was super annoying to try to make work.  primarily because a lot of these values need to be pushed through to the `: base(...)` call.  SO i'm going to keep as is.